### PR TITLE
Feature/calico cni

### LINKED ISSUE #75: Ensure we are supporting K8s NetPols
By default, EKS ships with it's own VPC-CNI, but it currently [doesn't support K8s NetworkPolicy resources](https://github.com/aws/containers-roadmap/issues/1478). For this EKS module, let's make sure that K8s NetPols can be implemented and enforced. This [Github issue](https://github.com/aws/amazon-vpc-cni-k8s/issues/1257#issuecomment-706408142) recommends Calico or Security Groups for Pods, but each of those comes with its own pros/cons.

### LINKED ISSUE #149: create_aws_auth_configmap should be a boolean input variable
Ran into a bug around the eks module's local for `create_aws_auth_configmap` logic. 
Simplest solution: make it a boolean var like the upstream.

Kind of a strange use case, but our module should be modular.

Error replication:
1. Deploy managed and self managed node groups
2. Try to destroy just the managed node groups by emptying the managed node group map. Scenario would be to destroy the managed node groups, deploy new ones, whatever.
  - This then essentially changes the logic around `create_aws_auth_configmap`, sets it to true, and tries to _create it_ in the upstream module. The issue here is, `aws-auth` configmap already exists due to previously deploying eks managed node groups on cluster instantiation 
3. receive error:
│ Error: configmaps “aws-auth” already exists
│ 
│   with module.eks.module.aws_eks.kubernetes_config_map.aws_auth[0],
│   on .terraform/modules/eks.aws_eks/main.tf line 536, in resource “kubernetes_config_map” “aws_auth”:
│  536: resource “kubernetes_config_map” “aws_auth” {

### LINKED ISSUE #153: Need to set tags better
Recently we've been facing issues cleaning up the account with orphaned pipeline TF resources.
A solution to fix this would be to standardize the tags deployed on resources when running the pipeline.

This would make it easier to run aws-nuke and target things that either have a name or specific tags and lower the overall blast radius.

### PR DESCRIPTION
**Pull request by zack-is-cool**
closes #75  
resolves #149
resolves #153 
resloves #124 

Calico and vpc-cni general blueprints
- Add calico to support network policies
- Turn on the rest of the blueprints add ons we were using before
- Fix add compatibility for vpc-cni and calico w/ `"ANNOTATE_POD_IP": "true"`
- Simplify `cluster_addons` input (until we have a need to template it for whatever reason (functions, vars as inputs for the config, etc))
  - This was to have parity with upstream as an input
  
  
create_aws_auth_configmap should be a boolean input variable. This logic is flakey and is easier to manage from just setting a boolean.

Setting tags "better" - Added provider tags and ensured tags are applied to child modules in a similar pattern throughout. Use-case for this is to simplify tagging on resources for targeted aws-nuke operations based on either names, or tags if the objects have non-standard identifiers pulled from aws-nuke (i.e security groups, vpcs, ec2 instances, etc).

Other changes 
- changed tenancy to default for the `secure` test to save $$$
- update aws provider version

### CODE REVIEW

File: modules/eks/locals.tf [hunk 1]
```diff
@@ -8,17 +8,11 @@ locals {
...
     [data.aws_caller_identity.current.arn]
   ))
-  aws_auth_users = distinct(concat([for admin_user in var.aws_admin_usernames : {
+  aws_auth_users = [for admin_user in var.aws_admin_usernames : {
     userarn  = "arn:${data.aws_partition.current.partition}:iam::${data.aws_caller_identity.current.account_id}:user/${admin_user}"
     username = admin_user
     groups   = ["system:masters"]
-    }],
-    [{
-      userarn  = data.aws_caller_identity.current.arn
```
Comment (zack-is-cool) [inline modules/eks/locals.tf:17]: Removing this because this is redundant. The current (making the cluster) is always added (invisibly in aws-auth)

**Comment by zack-is-cool** [comment]
/test all
Now with helm not waiting.. maybe no timeout

**Comment by zack-is-cool** [comment]
/test all
tags?

**Comment by zack-is-cool** [comment]
/test all
provider tags are bad with merging/applying other tags

**Comment by zack-is-cool** [comment]
/test all
RDS timeout 🙄

File: modules/bastion/s3-buckets.tf [hunk 2]
```diff
@@ -104,6 +104,10 @@ resource "aws_s3_bucket_logging" "access_logging_on_session_logs_bucket" {
 
   target_bucket = aws_s3_bucket.access_log_bucket.id
   target_prefix = "log/"
+  depends_on = [
+    aws_s3_bucket.access_log_bucket,
+    aws_s3_bucket.session_logs_bucket
+  ]
 }
```
Comment (zack-is-cool) [inline modules/bastion/s3-buckets.tf:111]: I had to add this for the `empty object` errors - I think it's really dependent on the `target_bucket`

File: examples/complete/fixtures.secure.tfvars [hunk 3]
```diff
@@ -2,6 +2,8 @@ region                         = "us-east-2"
 region2                        = "us-east-1"
 enable_eks_managed_nodegroups  = false
 enable_self_managed_nodegroups = true
-bastion_tenancy                = "dedicated"
-eks_worker_tenancy             = "dedicated"
+bastion_tenancy                = "default"
+eks_worker_tenancy             = "default"
```
Comment (ntwkninja) [inline examples/complete/fixtures.secure.tfvars:6]: I'm not a big fan of us not testing this. Did you get feedback about cost?

**Review by ntwkninja — COMMENTED** [review]
I'm ok with approving this PR once the comment is resolved; however, please try to keep future PRs to a more narrow focus (i.e. doesn't contain several unrelated issues in the same PR). It makes the review much easier when the scope is reduced.

File: examples/complete/fixtures.secure.tfvars (hunk 3 above)
Comment (zack-is-cool) [reply examples/complete/fixtures.secure.tfvars:6]: No, it was more of just an idea that using dedicated vs default didn't really test much using the terraform and it was cheaper. Can change it back if you'd rather us test it.

Comment (ntwkninja) [reply examples/complete/fixtures.secure.tfvars:6]: I couldn't find any edge cases that would have been concerning from a testing perspective but the only way we'll know is if we test it as we anticipate it being used.

Comment (zack-is-cool) [reply examples/complete/fixtures.secure.tfvars]: Reverted

**Review by ntwkninja — APPROVED** [review]
⚡
